src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +47,443 @@ public ByteArraySet createSmallSet() {
         return set;
     }
 
+    public List<ByteArray> getRandomByteArrays(int count) {
+        List<ByteArray> vals = new ArrayList<>();
+        while(vals.size() < count) {
+            String val = RandomStringUtils.randomAlphanumeric(6);
+            if (!vals.contains(val)) {
+                vals.add(new ByteArray(val));
+            }
+        }
+        Collections.shuffle(vals);
+        return vals;
+    }
+
+    public void testAdd(int size) {
```

Comment:
I added tests around the public methods of `ByteArraySet` to see if I could locate the bug, which without success.